bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestDataClientVeneerApi.java

```diff
@@ -242,15 +395,56 @@ public void testReadRowsCancel() throws IOException {
     when(mockIter.next()).thenReturn(EXPECTED_RESULT);
 
     ResultScanner resultScanner = dataClientWrapper.readRows(query);
-    assertResult(EXPECTED_RESULT, resultScanner.next());
+    new Thread(
+            () -> {
+              try {
+                assertResult(EXPECTED_RESULT, resultScanner.next());
+              } catch (IOException e) {
+                fail(String.valueOf(e));
+              }
+            })
+        .start();
 
     doNothing().when(serverStream).cancel();
     resultScanner.close();
 
-    // make sure that the scanner doesn't iteract with the iterator on close
+    // make sure that the scanner doesn't interact with the iterator on close
     verify(serverStream).cancel();
-    verify(mockIter, times(1)).hasNext();
-    verify(mockIter, times(1)).next();
+  }
+
+  @Test
+  public void testRead100Rows() throws IOException {
```

Comment:
Was this a new test that's added to test the paginator?

Reply:
This test isnt really testing anything, I would just drop it

Reply:
Done